SpringProject/src/it/java/com/gradproject2019/AuthControllerIT.java

```diff
@@ -0,0 +1,122 @@
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static org.springframework.http.HttpMethod.POST;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment= SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class AuthControllerIT {
+
+    @Autowired
+    private AuthRepository authRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @LocalServerPort
+    int testServerPort;
+
+    private User user;
+    private String realPassword;
+    private String baseUrl;
+
+    @Before
+    public void setUp() {
+        authRepository.deleteAll();
+        userRepository.deleteAll();
+        realPassword = PasswordUtils.hash("P455w0rd!");
+        user = new User(1L, "KaramsCoolUsername", "Karam", "Kapoor", "KSinghK@gmail.com", realPassword, "Botanist");
+        baseUrl = "http://localhost:" + testServerPort + "/auth";
+        restTemplate.getRestTemplate().setRequestFactory(new HttpComponentsClientHttpRequestFactory());
+    }
+
+    @After
+    public void tearDown() {
+        authRepository.deleteAll();
+        userRepository.deleteAll();
+    }
+
+    @Test
+    public void shouldReturn200andCreateTokenWhenUserExistsAndPasswordCorrect() throws URISyntaxException {
+        //given
+        URI uri = new URI(baseUrl +"/login");
+        LoginDto loginDto = createLoginDto("KaramsCoolUsername", "P455w0rd!");
+        HttpEntity<LoginDto> request = new HttpEntity<>(loginDto);
+
+        //when
+        User savedUser = userRepository.saveAndFlush(user);
+        ResponseEntity<Token> response = login(uri, request);
+
+        //then
+        Assert.assertEquals(200, response.getStatusCodeValue());
+        Assert.assertEquals(savedUser.getId(), response.getBody().getUserId());
+    }
+
+    @Test
+    public void shouldReturn404WhenUserDoesNotExists() throws URISyntaxException{
+        //given
+        URI uri = new URI(baseUrl +"/login");
+        LoginDto loginDto = createLoginDto("KaramsCoolUsername", "P455w0rd!");
+        HttpEntity<LoginDto> request = new HttpEntity<>(loginDto);
+
+        //when
+        ResponseEntity<Token> response = login(uri, request);
+
+        //then
+        Assert.assertEquals(404, response.getStatusCodeValue());
+    }
+
+    @Test
+    public void shouldReturn403nWhenUserExistsAndPasswordIncorrect() throws URISyntaxException {
+        //given
+        URI uri = new URI(baseUrl +"/login");
+        LoginDto loginDto = createLoginDto("KaramsCoolUsername", "WrongPassword");
+        HttpEntity<LoginDto> request = new HttpEntity<>(loginDto);
+
+        //when
+        userRepository.saveAndFlush(user);
+        ResponseEntity<Token> response = login(uri, request);
+
+        //then
+        Assert.assertEquals(403, response.getStatusCodeValue());
```

Comment:
This test should also assert on the message returned (if in fact one is being returned to the client, and if this is not the case, then why?)

Reply:
This also applies to the user not found test above this test in the same file.

Reply:
This also applies to the user not found test above this one.